mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/HighAvailabilityServicesUtil.java

```diff
@@ -182,9 +182,14 @@ public void register(ResourceLeaderChangeListener<ResourceClusterGateway> change
               .observeOn(scheduler)
               .subscribe(nextDescription -> {
                 log.info("nextDescription={}", nextDescription);
+
+                ResourceClusterGatewayClient newResourceClusterGatewayClient = new ResourceClusterGatewayClient(clusterID, nextDescription);
```

Comment:
It would be more appropriate to compare the descriptions rather than the clients themselves in terms of semantics. I’m concerned that if the logic for clients being equal changes in the future, it could cause problems for this logic.